### PR TITLE
partial rename from Service -> Manifest for groups of k8s objects

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @jazzdan,

Please review the following commits I made in branch nicks/ch55/service:

01a747d1f81d58bd74982187906f3882d27832bd (2018-09-13 14:56:44 -0400)
partial rename from Service -> Manifest for groups of k8s objects

### CODE REVIEW

**Comment by nicks** [comment]
(there are still a lot of uses of 'service', and going to eventually get all of them, but this is just a start)

File: internal/cli/cli.go [hunk 1]
```diff
@@ -68,6 +68,6 @@ func addCommand(parent *cobra.Command, child tiltCmd) {
 	parent.AddCommand(cobraChild)
 }
 
-func provideServiceCreator(upper engine.Upper, sm service.Manager) model.ServiceCreator {
+func provideServiceCreator(upper engine.Upper, sm service.Manager) model.ManifestCreator {
```
Comment (maiamcc) [inline internal/cli/cli.go]: manifest creator?

File: internal/cli/wire.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/windmilleng/tilt/internal/service"
 )
 
-func wireServiceCreator(ctx context.Context, browser engine.BrowserMode) (model.ServiceCreator, error) {
+func wireServiceCreator(ctx context.Context, browser engine.BrowserMode) (model.ManifestCreator, error) {
```
Comment (maiamcc) [inline internal/cli/wire.go]: wireManifestCreator

File: internal/engine/build_and_deployer.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ type BuildAndDeployer interface {
 	//
 	// BuildResult can be used to construct a BuildState, which contains
 	// the last successful build and the files changed since that build.
-	BuildAndDeploy(ctx context.Context, service model.Service, currentState BuildState) (BuildResult, error)
+	BuildAndDeploy(ctx context.Context, service model.Manifest, currentState BuildState) (BuildResult, error)
```
Comment (maiamcc) [inline internal/engine/build_and_deployer.go]: change arg name `service` --> `manifest`

File: internal/engine/image_build_and_deployer.go [hunk 4]
```diff
@@ -72,7 +72,7 @@ func (ibd *ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, service mo
 	}, nil
 }
 
-func (ibd *ImageBuildAndDeployer) build(ctx context.Context, service model.Service, state BuildState) (reference.NamedTagged, error) {
+func (ibd *ImageBuildAndDeployer) build(ctx context.Context, service model.Manifest, state BuildState) (reference.NamedTagged, error) {
```
Comment (maiamcc) [inline internal/engine/image_build_and_deployer.go]: change arg names here as well

File: internal/engine/image_build_and_deployer.go [hunk 5]
```diff
@@ -117,7 +117,7 @@ func (ibd *ImageBuildAndDeployer) build(ctx context.Context, service model.Servi
 	return n, nil
 }
 
-func (ibd *ImageBuildAndDeployer) deploy(ctx context.Context, service model.Service, n reference.NamedTagged) ([]k8s.K8sEntity, error) {
+func (ibd *ImageBuildAndDeployer) deploy(ctx context.Context, service model.Manifest, n reference.NamedTagged) ([]k8s.K8sEntity, error) {
```
Comment (maiamcc) [inline internal/engine/image_build_and_deployer.go]: arg names

File: internal/engine/testdata_test.go [hunk 6]
```diff
@@ -39,7 +39,7 @@ FROM go:1.10
 
 var SanchoTag, _ = reference.ParseNormalizedNamed("gcr.io/some-project-162817/sancho")
 
-var SanchoService = model.Service{
+var SanchoService = model.Manifest{
```
Comment (maiamcc) [inline internal/engine/testdata_test.go]: SanchoManifest

File: internal/engine/up_watch.go [hunk 7]
```diff
@@ -12,7 +12,7 @@ import (
 )
 
 type serviceFilesChangedEvent struct {
```
Comment (maiamcc) [inline internal/engine/up_watch.go]: manifestFilesChangedEvent?

File: internal/engine/up_watch.go [hunk 8]
```diff
@@ -12,7 +12,7 @@ import (
...
-	service model.Service
+	service model.Manifest
```
Comment (maiamcc) [inline internal/engine/up_watch.go]: `manifest model.Manifest`

File: internal/engine/upper.go [hunk 9]
```diff
@@ -199,7 +199,7 @@ func (u Upper) logBuildEvent(ctx context.Context, service model.Service, buildSt
 	logger.Get(ctx).Infof("Rebuilding service: %s", service.Name)
```
Comment (maiamcc) [inline internal/engine/upper.go]: "rebuilding manifest"

File: internal/engine/upper_test.go [hunk 10]
```diff
@@ -108,21 +108,21 @@ func TestUpper_Up(t *testing.T) {
 	f := newTestFixture(t)
 	defer f.TearDown()
 	service := f.newService("foobar", nil)
```
Comment (maiamcc) [inline internal/engine/upper_test.go]: 👀

File: internal/engine/upper_test.go [hunk 11]
```diff
@@ -108,21 +108,21 @@ func TestUpper_Up(t *testing.T) {
...
-	err := f.upper.CreateServices(output.CtxForTest(), []model.Service{service}, false)
+	err := f.upper.CreateManifests(output.CtxForTest(), []model.Manifest{service}, false)
 	close(f.b.calls)
 	assert.Nil(t, err)
-	var startedServices []model.Service
+	var startedServices []model.Manifest
```
Comment (maiamcc) [inline internal/engine/upper_test.go]: startedManifests

File: internal/model/manifest.go [hunk 12]
```diff
@@ -8,50 +8,50 @@ import (
...
 	}
 	return nil
 }
 
-func (s Service) validate() *ValidateErr {
-	if s.Name == "" {
-		return validateErrf("[validate] service missing name: %+v", s)
+func (m Manifest) validate() *ValidateErr {
+	if m.Name == "" {
+		return validateErrf("[validate] service missing name: %+v", m)
```
Comment (maiamcc) [inline internal/model/manifest.go]: change error strings

File: internal/service/creator.go [hunk 13]
```diff
@@ -8,16 +8,16 @@ import (
 
 // A creator that adds all new services to the manager.
```
Comment (maiamcc) [inline internal/service/creator.go]: "all new manifests"

File: internal/service/creator.go [hunk 14]
```diff
@@ -8,16 +8,16 @@ import (
...
 }
 
-func TrackServices(delegate model.ServiceCreator, manager Manager) trackingCreator {
+func TrackServices(delegate model.ManifestCreator, manager Manager) trackingCreator {
 	return trackingCreator{delegate: delegate, manager: manager}
 }
 
-func (c trackingCreator) CreateServices(ctx context.Context, svcs []model.Service, watch bool) error {
-	err := c.delegate.CreateServices(ctx, svcs, watch)
+func (c trackingCreator) CreateManifests(ctx context.Context, svcs []model.Manifest, watch bool) error {
```
Comment (maiamcc) [inline internal/service/creator.go]: arg name

File: internal/service/manager.go [hunk 15]
```diff
@@ -8,28 +8,28 @@ import (
...
+	Update(s model.Manifest) error
+	Remove(s model.ManifestName)
+	List() []model.Manifest
+	Get(n model.ManifestName) (model.Manifest, error)
 }
 
 type memoryManager struct {
 	mu       *sync.Mutex
-	services map[model.ServiceName]model.Service
+	services map[model.ManifestName]model.Manifest
```
Comment (maiamcc) [inline internal/service/manager.go]: field name

File: internal/service/manager.go [hunk 16]
```diff
@@ -55,16 +55,16 @@ func (m *memoryManager) Update(s model.Service) error {
 	return nil
 }
 
-func (m *memoryManager) serviceExists(n model.ServiceName) bool {
+func (m *memoryManager) serviceExists(n model.ManifestName) bool {
 	_, servicePresent := m.services[n]
```
Comment (maiamcc) [inline internal/service/manager.go]: manifestPresent

File: internal/service/manager_test.go [hunk 17]
```diff
@@ -12,21 +12,21 @@ import (
 func TestAdd(t *testing.T) {
 	f := newServiceManagerFixture(t)
 
-	s := model.Service{Name: model.ServiceName("hello")}
+	s := model.Manifest{Name: model.ManifestName("hello")}
```
Comment (maiamcc) [inline internal/service/manager_test.go]: should probably change alllll the `s` variables to be `m`

File: internal/service/manager_test.go [hunk 18]
```diff
@@ -140,6 +140,6 @@ func newServiceManagerFixture(t *testing.T) *serviceManagerFixture {
 	}
 }
 
-func (f *serviceManagerFixture) AssertServiceList(s []model.Service) {
+func (f *serviceManagerFixture) AssertServiceList(s []model.Manifest) {
```
Comment (maiamcc) [inline internal/service/manager_test.go]: s/serviceManagerFixture/manifestManagerFixture, and also arg name

File: internal/tiltfile/tiltfile.go [hunk 19]
```diff
@@ -172,7 +172,7 @@ func (tiltfile Tiltfile) GetServiceConfigs(serviceName string) ([]model.Service,
 
 	switch service := val.(type) {
 	case compService:
-		var servs []model.Service
+		var servs []model.Manifest
```
Comment (maiamcc) [inline internal/tiltfile/tiltfile.go]: var name

File: internal/tiltfile/tiltfile.go [hunk 20]
```diff
@@ -188,33 +188,33 @@ func (tiltfile Tiltfile) GetServiceConfigs(serviceName string) ([]model.Service,
...
+		s.Name = model.ManifestName(serviceName)
+		return []model.Manifest{s}, nil
 
 	default:
 		return nil, fmt.Errorf("'%v' returned a '%v', but it needs to return a k8s_service or composite_service", serviceName, val.Type())
 	}
 }
 
-func skylarkServiceToDomain(service k8sService) (model.Service, error) {
+func skylarkServiceToDomain(service k8sService) (model.Manifest, error) {
```
Comment (maiamcc) [inline internal/tiltfile/tiltfile.go]: arg name maybe?

**Comment by nicks** [comment]
yep! i appreciate the comments, but i'm going to do those on a separate PR

Find -> Replace on all files is easier than trying to do selective find and replace

**Pull request merged**